R/tinytheme.R

```diff
@@ -0,0 +1,153 @@
+#' Set or Reset Plot Themes for `tinyplot`
+#'
+#' @description
+#' The `tinytheme` function sets or resets the theme for plots created with
+#' `tinyplot`. Themes control the appearance of plots, such as text alignment,
+#' font styles, and axis labels. By default, a "bw" theme is available.
+#'
+#' Warning: `tinytheme()` shuts down all open graphical devices.
+#'
+#' @param theme A character string specifying the name of the theme to apply.
+#'   - `"default"`
+#'   - `"bw"`
+#'   - `"classic"`
+#'   - `"dark"`
+#'   - `"ipsum"`
+#'   - `"minimal"`
+#' @param ... Named arguments to override specific theme settings. These arguments are
+#'   passed to `tpar()` and take precedence over the predefined settings in the selected
+#'   theme.
+#'
+#' @details
+#' Sets a list of graphical parameters using `tpar()`
+#'
+#' To reset the theme to default settings (no customization), call `tinytheme()` without arguments.
+#'
+#' The `"custom"` theme sets parameters stored in a list in a global option. See examples below.
+#'
+#' @return The function returns nothing. It is called for its side effects.
+#'
+#' @examples
+#' # Set a theme
+#' tinytheme("bw")
+#' tinyplot(mpg ~ hp | factor(am), data = mtcars)
+#'
+#' # Customize the theme by overriding default settings
+#' tinytheme("bw", fg = "blue", font.main = 2)
+#' tinyplot(mpg ~ hp | factor(am), data = mtcars, main = "Hello World!")
+#'
+#' # Reset the theme
+#' tinytheme()
+#' tinyplot(mpg ~ hp | factor(am), data = mtcars)
+#'
+#' # Custom theme
+#' tinytheme("default", font.main = 3, col = "red")
+#' tinyplot(mpg ~ hp | factor(am), data = mtcars)
+#'
+#' @export
+tinytheme = function(theme = "default", ...) {
+  # in interactive sessions, we close the graphics device to reset parameters
+  if (interactive()) {
+    grDevices::graphics.off()
+  } else {
```

Comment:
Note to self: I think we can get away without having to shut the device by explicitly specifying the parameters of the "default" case. (I _really_ want to avoid shutting the device and I don't think CRAN will let us do it either...)